src/plugins/info/plugin.py

```diff
@@ -237,6 +375,81 @@ def _size(value, factor = 1):
             return f"{color}{value:3.1f} {unit}"
         value /= 1000.0
 
+# To validate if a file is within the file tree,
+# it needs to be absolute, so that it is possible to
+# check the prefix.
+def _convert_to_abs(path: str, abs_prefix: str = None) -> str:
+    if os.path.isabs(path): return path
+    if abs_prefix is None: abs_prefix = os.getcwd()
+    return os.path.normpath(os.path.join(abs_prefix, path))
+
+# Custom YAML loader - required to handle the parent INHERIT config.
+# It converts the INHERIT path to absolute as a side effect.
+# Returns the loaded config, or a list of all loaded configs.
+def _load_yaml(abs_src_path: str):
```

Comment:
From what I read here, it is not apparent to me why we need to implement this ourselves and can't just use MkDocs loading functionality. It looks like a duplication of. It is a good idea to explain this in the comment.

Example:

https://github.com/squidfunk/mkdocs-material/blob/15f15068d958506734d3544108a1e6971d7e511b/src/plugins/blog/structure/__init__.py#L59-L75